diff --git a/launcher/Application.cpp b/launcher/Application.cpp
index db3487726..66609c435 100644
--- a/launcher/Application.cpp
+++ b/launcher/Application.cpp
@@ -637,7 +637,7 @@ Application::Application(int& argc, char** argv) : QApplication(argc, argv)
         // Native library workarounds
         m_settings->registerSetting("UseNativeOpenAL", false);
         m_settings->registerSetting("CustomOpenALPath", "");
-        m_settings->registerSetting("UseNativeGLFW", false);
+        m_settings->registerSetting("UseNativeGLFW", true);
         m_settings->registerSetting("CustomGLFWPath", "");
 
         // Peformance related options
