diff --git a/Rakefile b/Rakefile
index 9deb877..b0200bf 100644
--- a/Rakefile
+++ b/Rakefile
@@ -17,7 +17,7 @@ begin
     gem.summary = %Q{Simple and easy way to know if you're on windows or not (reliably), as well as how many bits the OS is, etc.}
     gem.description = %Q{The OS gem allows for some useful and easy functions, like OS.windows? (=> true or false) OS.bits ( => 32 or 64) etc"}
     gem.email = "rogerpack2005@gmail.com"
-    gem.homepage = "http://github.com/rdp/os"
+    gem.homepage = "https://github.com/rdp/os"
     gem.authors = ["rdp", "David McCullars"]
     gem.license = "MIT"
     # gem is a Gem::Specification... see http://www.rubygems.org/read/chapter/20 for additional settings
diff --git a/os.gemspec b/os.gemspec
index 63d3d79..77c26b9 100644
--- a/os.gemspec
+++ b/os.gemspec
@@ -37,7 +37,7 @@ Gem::Specification.new do |s|
     "spec/osx_spec.rb",
     "spec/spec_helper.rb"
   ]
-  s.homepage = "http://github.com/rdp/os".freeze
+  s.homepage = "https://github.com/rdp/os".freeze
   s.licenses = ["MIT".freeze]
   s.rubygems_version = "2.7.6".freeze
   s.summary = "Simple and easy way to know if you're on windows or not (reliably), as well as how many bits the OS is, etc.".freeze

